source/codegen/templates/service_helpers.mako

```diff
@@ -101,41 +101,48 @@ ${initialize_standard_input_param(function_name, parameter)}\
 </%def>
 
 ## Initialize an enum input parameter for an API call.
-<%def name="initialize_enum_input_param(function_name, parameter)">\
+<%def name="initialize_enum_input_param(function_name, parameter, enums)">\
 <%
+  config = data['config']
+  namespace_prefix = config["namespace_component"] + "_grpc::"
   parameter_name = common_helpers.camel_to_snake(parameter['cppName'])
-  pascal_parameter_name = common_helpers.snake_to_pascal(parameter_name)
+  field_name = common_helpers.camel_to_snake(parameter["name"])
+  pascal_field_name = common_helpers.snake_to_pascal(field_name)
   map_name = parameter["enum"].lower() + "_input_map_"
-  iterator_name = parameter_name + "_imap_it"
-  enum_type_prefix = function_name + "Request::" + pascal_parameter_name + "EnumCase::"
-  param_all_caps_snake = parameter_name.upper()
+  iterator_name = field_name + "_imap_it"
+  one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{pascal_field_name}EnumCase'
+  enum_request_snippet = f'request->{field_name}()'
+  raw_request_snippet = f'request->{field_name}_raw()'
 %>\
       ${parameter['type']} ${parameter_name};
-      switch (request->${parameter_name}_enum_case()) {
-        case ${enum_type_prefix}k${pascal_parameter_name}: {
-          auto ${iterator_name} = ${map_name}.find(request->${parameter_name}());
+      switch (request->${field_name}_enum_case()) {
+        case ${one_of_case_prefix}::k${pascal_field_name}: {
+% if enums[parameter["enum"]].get("generate-mappings", False):
+          auto ${iterator_name} = ${map_name}.find(${enum_request_snippet});
           if (${iterator_name} == ${map_name}.end()) {
             return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The value for ${parameter_name} was not specified or out of range.");
           }
+<%
+enum_request_snippet = f'{iterator_name}->second'
+%>
+% endif
 % if parameter['type'] in ["ViConstString", "ViString"]:
-          ${parameter_name} = const_cast<${parameter['type']}>((${iterator_name}->second).c_str());
+          ${parameter_name} = const_cast<${parameter['type']}>((${enum_request_snippet}).c_str());
 % else:
-          ${parameter_name} = static_cast<${parameter['type']}>(${iterator_name}->second);
+          ${parameter_name} = static_cast<${parameter['type']}>(${enum_request_snippet});
 % endif
           break;
         }
-        case ${enum_type_prefix}k${pascal_parameter_name}Raw: {
+        case ${one_of_case_prefix}::k${pascal_field_name}Raw:
```

Comment:
Why keep the `{` on the first switch case but remove it from this one? I recommend putting the `{` and `}` back in.

Reply:
Updated to add `{ }` for all cases in switch.